File: III-sem/java/JavaLabPrograms/factorial.java
//
//factorial.java
//
//Program to compute factorial of n
//
import java.util.Scanner;
import java.math.BigInteger; 

public class factorial
{
    public BigInteger factorialOfN(int n)
    {
        BigInteger fact = new BigInteger("1");
        for( int i = n ; i > 1 ; i--)
            fact = fact.multiply(BigInteger.valueOf(i));
        return fact;
    }
    public static void main(String []args)
    {
        int n;
        Scanner scan = new Scanner(System.in);
        factorial obj = new factorial();

        System.out.print("Enter value of n : ");
        n = scan.nextInt();

        System.out.println("Factorial of " + n + " is : " + obj.factorialOfN(n));
    }
}

File: Misc/WORKSHOPS/Data Structures & Algorithms by Kartik Mathur from Coding Blocks on 20-08-19/factorial.cpp
//
//factorial.cpp
//
//program to compute factorial of a number
//
#include<iostream>
using namespace std;

long int factorial(long int n)
{
    if(n == 0 || n == 1)
        return n;
    else
     return n * factorial(n - 1);
}

int main()
{
    int n;

    cout << "Enter value of n  : ";
    cin >> n;

    cout << "Factorial of " << n << " is : " << factorial(n);
    
    return 0;
}